### PR TITLE
Bug 1949589: Add APIRemovedInNextEUSReleaseInUse Alert

### PR DESCRIPTION
**Pull request by deads2k**
updated version of https://github.com/openshift/cluster-kube-apiserver-operator/pull/1094 to reflect new API and use static-resource-controller instead of CVO

builds on https://github.com/openshift/cluster-kube-apiserver-operator/pull/1073 to avoid conflicts

### CODE REVIEW

File: bindata/v4.1.0/alerts/kube-apiserver-requests.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+metadata:
+  name: kube-apiserver-requests
+  namespace: openshift-kube-apiserver
+  annotations:
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+    exclude.release.openshift.io/internal-openshift-hosted: "true"
+spec:
+  groups:
+    - name: apiserver-requests-in-flight
```
Comment (sttts) [inline bindata/v4.1.0/alerts/kube-apiserver-requests.yaml:12]: where does this come from?

File: bindata/v4.1.0/alerts/api-usage.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+metadata:
+  name: api-usage
+  namespace: openshift-kube-apiserver
+  annotations:
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+    exclude.release.openshift.io/internal-openshift-hosted: "true"
+spec:
+  groups:
+    - name: pre-release-lifecycle
```
Comment (sttts) [inline bindata/v4.1.0/alerts/api-usage.yaml:12]: this move will break the e2e test checking the version in the expression

**Comment by sttts** [comment]
I disagree with "use static-resource-controller instead of CVO". We shouldn't do this. It creates complexity.

File: bindata/v4.1.0/alerts/api-usage.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: api-usage
+  namespace: openshift-kube-apiserver
+  annotations:
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+    exclude.release.openshift.io/internal-openshift-hosted: "true"
```
Comment (sttts) [inline bindata/v4.1.0/alerts/api-usage.yaml:9]: those don't make sense anymore

**Comment by deads2k** [comment]
> I disagree with "use static-resource-controller instead of CVO". We shouldn't do this. It creates complexity.

our operator
1. reacts on a watch
2. produces events indicating status
3. sets operator status for the failing manifest
4. does a more intelligent diff
5. uses a lister cache instead of a GET

The CVO faces a watch fanout problem because of the namespace breadth.  In addition, it can never know the operator status itself.

The CVO is essential for getting the operator available and monitorable. Things that are inputs to the operator must be created by the CVO. But getting the operand functional is the responsibility of the operator. These files are for the operand, not the operator.

**Comment by deads2k** [comment]
/hold
Need to figure out why the metrics skip didn't work

**Comment by deads2k** [comment]
merged in https://github.com/openshift/cluster-kube-apiserver-operator/pull/1115

**Pull request closed without merging**